File: src/main/cpp/rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k %= nums.size();
        int begin = 0 - k + nums.size();
        if (begin == nums.size()) begin = 0;
        vector<int> res;
        for (int i = 0; i < nums.size(); ++i) {
            res.push_back(nums[begin]);
            ++begin;
            if (begin == nums.size()) begin = 0;
        }
        for (int i = 0; i < nums.size(); ++i) {
            nums[i] = res[i];
        }
    }
};

File: src/main/java/RotateArray.java
public class RotateArray {
    public void rotate(int[] nums, int k) {
        int[]res = new int[nums.length];
        int cnt = 0;
        k %= nums.length;
        int begin = 0 - k + nums.length;
        if (begin == nums.length) begin = 0;
        for (int i = 0; i < nums.length; ++i) {
            res[cnt++] = nums[begin++];
            if (begin == nums.length) begin = 0;
        }
        for (int i = 0; i < nums.length; ++i) nums[i] = res[i];
    }
}
